src/views/ActivityProfile/index.js

```diff
@@ -0,0 +1,174 @@
+import Button from 'material-ui/Button';
+import Card, { CardActions, CardContent } from 'material-ui/Card';
+import Email from 'material-ui-icons/Email';
+import Grid from 'material-ui/Grid';
+import IconButton from 'material-ui/IconButton';
+import List from 'material-ui/List';
+import Typography from 'material-ui/Typography';
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { withRouter } from 'react-router-dom';
+
+import activity from '../../services/activity';
+import './activity-profile.css';
+import GordonLoader from '../../components/Loader';
+import session from '../../services/session';
+
+class ActivityProfile extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      activityInfo: null,
+      // activityAdvisors: [],
+      activityFollowers: 0,
+      activityGroupAdmins: [],
+      activityMembers: 0,
+      activityStatus: null,
+      sessionInfo: null,
+    };
+  }
+
+  async componentWillMount() {
+    this.setState({ loading: true });
+    const { sessionCode, activityCode } = this.props.match.params;
+    const [activityInfo,
+      // activityAdvisors,
+      activityFollowers,
+      activityGroupAdmins,
+      activityMembers,
+      activityStatus,
+      sessionInfo] = await Promise.all([
+      activity.get(activityCode),
+      // activity.getAdvisors(activityCode, sessionCode),
+      activity.getFollowersNum(activityCode, sessionCode),
+      activity.getGroupAdmins(activityCode, sessionCode),
+      activity.getMembersNum(activityCode, sessionCode),
+      activity.getStatus(activityCode, sessionCode),
+      session.get(sessionCode),
+    ]);
+
+    this.setState({
+      activityInfo,
+      // activityAdvisors,
+      activityFollowers,
+      activityGroupAdmins,
+      activityMembers,
+      activityStatus,
+      sessionInfo,
+      loading: false,
+    });
+  }
+  render() {
+    let content;
+    if (this.state.loading === true) {
+      content = <GordonLoader />;
+    } else {
+      const { ActivityDescription: activityDescription } = this.state.activityInfo;
+      const { ActivityImagePath: activityImagePath } = this.state.activityInfo;
+      const followers = this.state.activityFollowers;
+      const members = this.state.activityMembers;
+      const { SessionDescription: sessionDescription } = this.state.sessionInfo;
+      let admins;
+      let groupContacts;
+      if (this.state.activityGroupAdmins.length > 0) {
+        groupContacts = (
```

Comment:
There are a few things that we could improve about this part of the component's code. First, the whole section could be combined into a single variable, which would remove the confusion of using both `groupContacts` and `admins` (it took me a while to figure out that they were two variables for the same section). Second, the `.map()` function is actually generating a whole independent `<List>` for each item, when it should instead be generating `<ListItem>`s for each item of the list, contained by a single `<List>` component. Third, the parameter passed to the `.map()` function's callback should be singular rather than plural, because it represents a single `activityGroupAdmin`, not multiple. Lastly, the `<List>` component has a `className` of `gordon-activity-profile`, which is redundant in this case because this class is already present on the `<CardContent>` component.

Here is how I would refactor it:

```JavaScript
      let groupContacts;
      if (this.state.activityGroupAdmins.length > 0) {
        groupContacts = (
          <Typography type="body1">
            <strong>Group Contacts:</strong>
            <List dense disablePadding>
              {this.state.activityGroupAdmins
                .map(activityGroupAdmin => (
                  <ListItem key={activityGroupAdmin.Email}>
                    &emsp;{activityGroupAdmin.FirstName} {activityGroupAdmin.LastName}
                    <IconButton
                      classes={({ root: 'email-button' })}
                      color="primary"
                      href={`mailto:${activityGroupAdmin.Email}`}
                      padding={0}
                    >
                      <Email
                        color="primary"
                        style={{ width: 16, height: 16 }}
                      />
                    </IconButton>
                  </ListItem>
                ))}
            </List>
          </Typography>
        );
      }
```

Can we change it to something like the above so the code communicates a bit more clearly?